app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/MovementService.scala

```diff
@@ -27,39 +31,69 @@ import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class MovementService @Inject()(
-                                 movementRepository: MovementRepository
-                                      )(implicit ec: ExecutionContext) {
-  def saveMovementMessage(movementMessage: Movement): Future[Either[GeneralMongoError, Movement]] = {
-    movementRepository.saveMovement(movementMessage)
-      .map(_ => Right(movementMessage))
-      .recover {
-        case ex: Throwable => Left(GeneralMongoError(ex.getMessage))
+                                 movementRepository: MovementRepository,
+                                 emcsUtils: EmcsUtils
+                               )(implicit ec: ExecutionContext) extends Logging {
+  def saveNewMovement(movement: Movement): Future[Either[Result, Movement]] = {
+
+    getMovementByLRNAndERNIn(movement.localReferenceNumber, List(movement.consignorId)).
+      flatMap {
+        case Some(movementFromDb: Movement) if movementFromDb.administrativeReferenceCode.isDefined || movementFromDb.consigneeId != movement.consigneeId =>
```

Comment:
ther line of code below
`if movementFromDb.administrativeReferenceCode.isDefined || movementFromDb.consigneeId != movement.consigneeId`

could be put in a function to explained what is doing. But not a blocker